modular_skyrat/master_files/code/modules/client/preferences/middleware/food.dm

```diff
@@ -0,0 +1,163 @@
+// To make this system not a massive meta-pick for gamers, while still allowing plenty of room for unique combinations.
+#define MINIMUM_REQUIRED_TOXICS 2
+#define MINIMUM_REQUIRED_DISLIKES 3
+#define MAXIMUM_LIKES 3
+
+// Performance and RAM friendly menu. You can thank me later.
+GLOBAL_DATUM_INIT(food_prefs_menu, /datum/food_prefs_menu, new)
+
+// Hahahaha, it LIVES!
+
+/datum/preference_middleware/food/apply_to_human(mob/living/carbon/human/target, datum/preferences/preferences)
+	if(!length(preferences.food) || isdummy(target))
+		return
+
+	var/fail_reason = GLOB.food_prefs_menu.is_food_invalid(preferences)
+	if(fail_reason)
+		to_chat(preferences.parent, span_doyourjobidiot("Your food preferences can't be set because of [fail_reason] choices! Please check your preferences!")) // Sorry, but I don't want folk sleeping on this.
+		return
+
+	var/datum/species/species = target.dna.species
+	species.liked_food = NONE
+	species.disliked_food = NONE
+	species.toxic_food = NONE
+
+	for(var/food_entry in GLOB.food_ic_flag_to_point_values)
+		var/list/food_points_entry = GLOB.food_ic_flag_to_point_values[food_entry]
+		var/food_preference = preferences.food[food_entry] || food_points_entry["[FOOD_DEFAULT]"]
+
+		switch(food_preference) // Doesn't use defines cause I need them in string form
+			if("1")
+				species.liked_food |= GLOB.food_ic_flag_to_bitflag[food_entry]
+			if("2")
+				species.disliked_food |= GLOB.food_ic_flag_to_bitflag[food_entry]
+			if("3")
+				species.toxic_food |= GLOB.food_ic_flag_to_bitflag[food_entry]
+
+/// Food prefs menu datum. Global datum for performance and memory reasons.
+/datum/food_prefs_menu/ui_interact(mob/dead/new_player/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "FoodPreferences", "Food Preferences")
+		ui.set_autoupdate(FALSE)
+		ui.open()
+
+/datum/food_prefs_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+
+	var/datum/preferences/preferences = ui?.user?.client?.prefs
+
+	switch(action)
+		if("reset")
+			qdel(preferences.food)
+			preferences.food = list()
+			return TRUE
+
+		if("toggle")
+			preferences.food["enabled"] = !preferences.food["enabled"]
+			return TRUE
+
+		if("change_food")
+
+			var/food_name = params["food_name"]
+			var/food_flag = params["food_flag"]
+
+			if(!food_name || !preferences || !food_flag || !(food_flag in list("[FOOD_LIKED]", "[FOOD_NEUTRAL]", "[FOOD_DISLIKED]", "[FOOD_TOXIC]")))
```

Comment:
It's not a flag if you're comparing it to strings, that completely defeats the purpose of flags.